tests/test_pages.py

```diff
@@ -102,18 +118,19 @@ def test_opening_magic_link_with_valid_token_redirects(client):
     url = reverse("magicauth-validate-token", args=[token.key])
     response = client.get(url)
     assert response.status_code == 302
-    assert response.url == "/test_home/"
+    assert response.url == "/landing/"
 
 
 def test_opening_magic_link_with_a_next_sets_a_new_url(client):
     token = factories.MagicTokenFactory()
-    url = (
-        reverse("magicauth-validate-token", kwargs={"key": token.key})
-        + "?next=/test_dashboard/?a=test&b=test"
-    )
+    next_url_raw = "/test_dashboard/?a=test&b=test"
+    # We use `quote` because the URL has parameters:
+    next_url = urllib.parse.quote(next_url_raw)
+    validate_token_url = reverse("magicauth-validate-token", kwargs={"key": token.key})
+    url = f'{validate_token_url}?next={next_url}'
     response = client.get(url)
     assert response.status_code == 302
-    assert response.url == "/test_dashboard/?a=test&b=test"
+    assert response.url == next_url_raw
 
 
 def test_opening_magic_link_with_a_unsafe_next_sets_triggers_404(client):
```

Comment:
Dans ce cas où le next est invalide, est-ce qu'on est logged in quand meme? Je dirais que oui, parce que ca pourrait arriver legitimement d'avoir un next-url invalide. (mais bon peut etre que c'est plus secure de bloquer le login ?)

Either way, je propose de le tester explicitement.